Lib/pdb.py

```diff
@@ -491,6 +491,8 @@ def interaction(self, frame, tb_or_exc):
                 Pdb._previous_sigint_handler = None
 
         _chained_exceptions, tb = self._get_tb_and_exceptions(tb_or_exc)
+        if not _chained_exceptions and isinstance(tb_or_exc, BaseException):
```

Comment:
I'm confused about the condition here - if ``tb_or_exc`` is a BaseException then ``_chained_exceptions`` should include at least this exception, no? 

And, don't we need to check tb before deciding there is no traceback?

Reply:
> I'm confused about the condition here - if `tb_or_exc` is a BaseException then `_chained_exceptions` should include at least this exception, no?

Not if `sys.last_exc = Exception()`, then `sys.last_exc.__traceback__` is None.

That is what the raise in the following was meant to prevent when passing a traceback:

```
def post_mortem(t=None):
    ...
    if t is None:
        exc = sys.exception()
        if exc is not None:
            t = exc.__traceback__

    if t is None:
        raise ValueError("A valid traceback must be passed if no "
                         "exception is being handled")

    ...
    p.interaction(None, t)

```

But as now you _can_ pass exceptions and they might be chained it's unsufficient.

> And, don't we need to check tb before deciding there is no traceback?

It's about the same, I find it clearer as it's

```
list_of_chain_exception, traceback_of_last_chained_exception = _get...
```

If `list_of_chain_exception`, then obviously `traceback_of_last_chained_exception` is empty,  but it's less obvious that `traceback_of_last_chained_exception = None` implies the whole list is empty.

If you prefer checking `tb is None`, I'm fine with it.

Reply:
added `and tb is None`

Reply:
> > I'm confused about the condition here - if `tb_or_exc` is a BaseException then `_chained_exceptions` should include at least this exception, no?
> 
> Not if `sys.last_exc = Exception()`, then `sys.last_exc.__traceback__` is None.

This is what I meant:

```
>>> pdb.Pdb()._get_tb_and_exceptions(Exception())
((Exception(),), None)
```

Reply:
> then obviously traceback_of_last_chained_exception is empty

Why is this obvious?

```
>>> try:
...    raise ValueError(42)
... except Exception as e:
...    exc = e
... 
>>> pdb.Pdb()._get_tb_and_exceptions(exc)
((ValueError(42),), <traceback object at 0x108dd2080>)
>>> pdb.Pdb()._get_tb_and_exceptions(exc)[0][-1].__traceback__
<traceback object at 0x108dd2080>
```

Reply:
Ok, I think I see what you're trying to do. But if you break when there is no ``__traceback__`` you remove this exception plus anything chained to it. There are places where tracebacks are filtered to exclude some frames (like in unittest). You can't assume that only the last exception in the chain can have an empty traceback.

Reply:
Ah, good point. I can just post-filter.